oauth2/oauth2-core/src/test/java/org/springframework/security/oauth2/core/ClaimAccessorTests.java

```diff
@@ -93,6 +93,17 @@ public void getClaimAsInstantWhenDoubleTypeSecondsThenReturnInstant() {
 				expectedClaimValue.minusSeconds(1), expectedClaimValue.plusSeconds(1));
 	}
 
+	// gh-6187
+	@Test
+	public void getClaimAsInstantWhenISODateStringThenReturnInstant() {
+		Instant expectedClaimValue = Instant.now();
```

Comment:
Let's explicitly provide the `String` value for the test, for example, `2011-12-03T10:15:30+01:00`